cpu/fe310/periph/spi.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2019 Tristan Bruns
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_fe310
+ * @ingroup     drivers_periph_spi
+ *
+ * @{
+ *
+ * @file        spi.c
+ * @brief       Low-level SPI driver implementation
+ *
+ * @author      Tristan Bruns
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "mutex.h"
+#include "assert.h"
+#include "periph/spi.h"
+#include "vendor/platform.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+/**
+ * @brief   Allocation device locks
+ */
+static mutex_t lock;
+
+void spi_init(spi_t bus)
+{
+    /* make sure given bus device is valid */
+    assert(bus == 0);
+
+    /* initialize the buses lock */
+    mutex_init(&lock);
+
+    /* trigger pin initialization */
+    spi_init_pins(bus);
+
+    /* disable hardware chip select
+       (hardware chip select only supports one-byte transfers...) */
+    SPI1_REG(SPI_REG_CSMODE) = SPI_CSMODE_OFF;
+}
+
+void spi_init_pins(spi_t bus)
+{
+    (void) bus;
```

Comment:
Should not needed because of the `assert(bus == 0);` below. (I believe both GCC and clang should still be aware that `bus` was used even when `NDEBUG` is defined, but I'm not 100% sure about that. So please correct me if I'm wrong.)